internal/gatewayapi/helpers.go

```diff
@@ -436,6 +436,16 @@ func irTLSConfigs(tlsSecrets ...*corev1.Secret) *ir.TLSConfig {
 			PrivateKey:  tlsSecret.Data[corev1.TLSPrivateKeyKey],
 		}
 	}
+
+	// Envoy Gateway defaults for ALPN:
+	// 1. HTTPS Routes: set IR to nil slice, which leads to xds-level
+	//    default of: h2, http/1.1. Envoy Proxy will support both HTTP2 and HTTP1.
+	// 2. Other xRoutes: set to empty list, disabling ALPN extension for Envoy Proxy.
+	//    ALPN cannot be used to negotiate application layer protocols like HTTP2.
+	if protocol != gwapiv1.HTTPSProtocolType {
+		tlsListenerConfigs.ALPNProtocols = []string{}
+	}
+
```

Comment:
Nit: we probably can make this easier to understand by:
 * Adding a line to the comments to explain the default behavior of the ALPNProtocols in the connection to the backend
 
 Or 
   * Adding another method 
   
   ```
   irBackendConnectionTLSConfigs(tlsSecrets ...*corev1.Secret) *ir.TLSConfig {
     return irTLSConfigs("", tlsSecrets)
   ```